src/test/java/org/jnode/fs/DataStructureAsserts.java

```diff
@@ -130,4 +132,83 @@ private static String toHexString(byte[] digest)
 
         return builder.toString();
     }
-}
+
+    /**
+     * Builds up the structure for the given file system entry to get the metadata.
+     *
+     * @param entry  the entry to process.
+     * @param actual the string to append to.
+     * @param indent the indent level.
+     * @throws IOException if an error occurs.
+     */
+    public static void buildXfsMetaDataStructure(FSEntry entry, StringBuilder actual, String indent) throws IOException
+    {
+        actual.append(indent);
+        actual.append(entry.getName());
+        actual.append("; \n");
+
+        if (entry.isDirectory()) {
+            getXfsMetadata(entry, actual, indent);
+        }
+        if (entry.isFile()) {
+            FSFile file = entry.getFile();
+            getXfsMetadata(entry, actual, indent);
+        }
+        else {
+            FSDirectory directory = entry.getDirectory();
+
+            Iterator<? extends FSEntry> iterator = directory.iterator();
+
+            while (iterator.hasNext()) {
+                FSEntry child = iterator.next();
+
+                if (".".equals(child.getName()) || "..".equals(child.getName()))
+                {
+                    continue;
+                }
+
+                buildXfsMetaDataStructure(child, actual, indent + "  ");
+            }
+        }
+    }
+
+    /**
+     * Get the metadata.
+     *
+     * @param entry  the entry to process.
+     * @param actual the string to append to.
+     * @param indent the indent level.
+     *
+     */
+    private static StringBuilder getXfsMetadata(FSEntry entry, StringBuilder actual,String indent)  throws IOException {
+        actual.append(indent);
+        actual.append(indent);
+        actual.append("atime : " +  getDate(((XfsEntry) entry).getLastAccessed() ));
```

Comment:
In `testXfsMetaData` It would be better to cast the return from `fs.getRootEntry()` to `XfsEntry` and get `buildXfsMetaDataStructure` and `getXfsMetaData` to take `XfsEntry` instead of `FSEntry`. This would save all the casts happening here. Card created. https://github.com/Nuix/jnode-fs/projects/1#card-76549508